### PR TITLE
Feat/turbo

### LINKED ISSUE #5: Move project to monorepo setup with turborepo
We can make usage of turborepo in the repo, and setup a monorepo with more than one package. Just so it makes our lives easier to publish e.g. the #6 

TODOs:

* [ ] ESLint setup
* [ ] Jest setup
* [ ] TS setup
* [ ] Turbo base pipelines setup
* [ ] NPM Workspaces setup

---

Things to consider:

- Will we have a `tsconfig.json` per package?
- Will we share ESLint config as a package or setup in the monorepo root?
- Will we keep most shared dependencies in the monorepo root?

### PR DESCRIPTION
**Pull request by sergioflores-j**
Added turbo to the project
Moved wrapException to a package
Added examples package, for more in-depth "real-world" scenarios. Can be used by the docs page.

Resolves #5

### CODE REVIEW

File: packages/wrapException/tsconfig.lib.json
Comment (sergioflores-j) [inline packages/wrapException/tsconfig.lib.json:1]: I added this tsconfig to simplify the build script

File: packages/wrapException/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@ts-to-go/wrap-exception",
+  "version": "1.0.0",
+  "description": "TS to go for better error handling",
+  "main": "dist/index.js",
+  "types": "dist/index.d.js",
+  "scripts": {
+    "build": "esbuild src/index.ts --bundle --platform=node --target=node16 --outfile=dist/index.js && npx --package=typescript tsc --project ./tsconfig.lib.json",
```
Comment (r5402) [inline packages/wrapException/package.json:8]: Can we make it more readable/cleaner?

File: turbo.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "build": {
+      "outputs": ["dist/**"]
+    },
+    "lint": {},
+    "test": {},
+    "type-check": {},
+    "publish": {}
```
[the remark this answers is not in the document]
Comment (r5402) [reply turbo.json:10]: One more pipeline and output directory for the docs. I'll add after we merge this one/open a PR for the docs.

**Review by r5402 — APPROVED** [review]

File: packages/wrapException/package.json (hunk 1 above)
Comment (sergioflores-j) [reply packages/wrapException/package.json:8]: Only if we write it in a makefile :/

**Pull request merged**